pkg/client/keycloak/config.go

```diff
@@ -110,6 +113,7 @@ func (kc *KeycloakConfig) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&kc.RedhatSSORealm.ClientIDFile, "redhat-sso-client-id-file", kc.RedhatSSORealm.ClientIDFile, "File containing Keycloak privileged account client-id that has access to the OSD Cluster IDP realm")
 	fs.StringVar(&kc.RedhatSSORealm.ClientSecretFile, "redhat-sso-client-secret-file", kc.RedhatSSORealm.ClientSecretFile, "File containing Keycloak privileged account client-secret that has access to the OSD Cluster IDP realm")
 	fs.StringVar(&kc.SsoBaseUrl, "redhat-sso-base-url", kc.SsoBaseUrl, "The base URL of the mas-sso, integration by default")
+	fs.StringVar(&kc.SpecialManagementOrgID, "special-management-org-id", SPEICAL_MGMT_ORG_ID_STAGE, "The Special Management Organization ID used for creating internal Service accounts")
```

Comment:
```suggestion
	fs.StringVar(&kc.SpecialManagementOrgID, "sso-special-management-org-id", SPEICAL_MGMT_ORG_ID_STAGE, "The Special Management Organization ID used for creating internal Service accounts")
```
the flag `special-management-org-id` might be too generic, I propose prefixing it with some thing that says it is a sso related management id. 

Questions:
1. This is meant to be used for the redhatsso_service.go right?
2. What do you think about start using it right away i.e all the internal service accounts are created with this information?

Reply:
1) Yes, post migration this information - is not needed to be sent in the request when creating service accounts, reason: we will be using a "ROOT" service account created in the special managment org (similar to the KFM client in MAS SSO). Since the root account is associated to the org, SSO takes care of associating the newly created internal service accounts to the appropriate org. This is needed as a future enhancement, for extra security for fleetshard communication back to KFM.
2) This can be done, but we deferred it since, that would cause an inconsistency for the already created service accounts. So migration tool has been updated to link to the correct orgID during the migration of internal service accounts from MAS SSO to sso.redhat.com

Reply:
For the Naming comment, Yes it makes sense, if we agree, the changes should be reflected in app-interface MR as well.

Reply:
> 1. Yes, post migration this information - is not needed to be sent in the request when creating service accounts, reason: we will be using a "ROOT" service account created in the special managment org (similar to the KFM client in MAS SSO). Since the root account is associated to the org, SSO takes care of associating the newly created internal service accounts to the appropriate org. This is needed as a future enhancement, for extra security for fleetshard communication back to KFM.

Why post migration? Won't we find ourselves with some internal accounts created without the special managed org id attribute until this work is completed? 

> 2. This can be done, but we deferred it since, that would cause an inconsistency for the already created service accounts. So migration tool has been updated to link to the correct orgID during the migration of internal service accounts from MAS SSO to sso.redhat.com

Thanks. This future work is tracked somewhere, correct?

Reply:
1) Currently in MAS SSO that is how it is. The reason for this in sso.redhat.com is that every service account (user or internal) should have a orgID associated so that it can be "MANAGED" via the API. So this is only required in sso.redhat.com env. 
2) This is already in place in the migration tool and we have verified it. The associated issue: https://issues.redhat.com/browse/MGDSTRM-7672

Reply:
I was asking for the special management organisation id to be used in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/14ae2b4dc33e4c3c7127dec35e703456b7e17241/pkg/services/sso/redhatsso_service.go or a issue/jira to track it and we are sure that it will be done before we start using that service file. 

But this is deferred for now until post migration, correct? So my question was: if we move to sso.redhat without making sure that the newly created internal service accounts are created under the special management id (using the redhatsso_service.go), won't we find ourselves with a gap there where some of the internal service accounts created post migration will be missing the special management org id value?

Reply:
No, that wont happen: When doing the migration: we will start using a new service account credential in vault (ssoredhat.com speical mgmt org root account). So there is no need to do this mapping. to compare: this root account is similar to "privileged KFM service account in MAS SSO" which can create/manage other SAs in this org. So when we switch to sso.redhat.com, we will switch to use a new SA (root SA for special mgmt org) to create service accounts. Service accounts created using this are mapped to the special mgmt org automatically/internally at sso.redhat.com and there is no need for adding that in the client side/SA representation as it is done on the server side (sso.redhat.com side).

Reply:
with the variable name, i have updated the code and every where to prefix with sso and updated the MR in app-interface as well

Reply:
Thanks for updating the flag name. 

I'll follow up on the other discussion privately. I feel like there is a piece of information that I am lacking and I want to be sure that we are not leaving off gapes.

Reply:
Sure Manyanda, lets discuss tomorrow

Reply:
Thanks @Rajagopalan-Ranganathan for the explanation.

> I feel like there is a piece of information that I am lacking and I want to be sure that we are not leaving off gapes.

We are all good here. What is going to happen is that the with sso.redhat.com, user facing service account creation will be using the user access token and for internal ones we'll be using the credentials. With the info on token, sso.redhat will be able to create the users service accounts into the user org and the internal ones in the internal org. This is the info I was missing.